src/utilities/analytics.ts

```diff
@@ -46,7 +52,21 @@ const mockGa = {
   // tslint:enable no-console
 };
 
-export const logAnalyticsEvent = (event: ILogEvent) => {
+export const logEvent = (runtimeContext: IPluginRuntimeContext, event: ILogEvent) => {
+  logAnalyticsEvent(event);
+  logPluginEvent(runtimeContext, event);
+};
+
+const logPluginEvent = (context: IPluginRuntimeContext, event: ILogEvent) => {
+  const eventTypeName = `TeacherEdition-${event.tipType}-${event.eventAction}`;
+  const logEventPayload: IPluginEventPayload = {
+    eventAction: event.eventAction,
+    eventContent: event.location ? event.location : event.tabName
+  };
+  context.log({ event: eventTypeName, event_value: logEventPayload });
```

Comment:
Are you sure you want to log that under `event_value` property?
I think this property is mostly used as a primitive value (or quite often not used at all). If you put an object here, it might be difficult to query logs using `eventAction` or `eventContent`, and these values won't look too good in CSV export.

If you just log them as a top level properties, they will probably go to `extras` hstore field which is expanded in CSV logs export and we could use its properties for queries. I think that our log system will simply put all the properties that it doesn't know in `extras`, e.g.:
```
context.log({ 
  event: eventTypeName, 
  eventAction: event.eventAction, 
  eventContent: event.location ? event.location : event.tabName
});
```

There's also a similar field called `parameters`, but you need to explicitly create it:
```
context.log({ 
  event: eventTypeName, 
  parameters: {
    eventAction: event.eventAction, 
    eventContent: event.location ? event.location : event.tabName
  }
});
```

What's the difference in practice? I don't know. But I think we use extras more often, as things just automatically go there.

btw, @scytacki or @dougmartin, do you remember if that's true with new log injection system too?
I guess it was handled by old rails app, so wondering if it's implemented in a similar way using Amazon stack. I don't remember what's the current status of this work and how far we are from switching to it.

Reply:
We figured out we didn't need to record both those props. So we're just going to include `eventContent` in `event_value`.